src/Angor/Shared/WalletOperations.cs

```diff
@@ -388,9 +476,19 @@ public async Task<(string address, List<UtxoData> data)> FetchUtxoForAddressAsyn
             if (utxo.Count < limit)
                 break;
 
+            if (allItems.Count >= maxutxo)
+            {
+                _logger.LogWarning($"utxo scan for address {address} was stopped after the limit of {maxutxo} was reached.");
+                break;
+            }
+                
+
             offset += limit;
         } while (true);
 
+        // todo: dan - this is a hack until the endpoint offset is fixed
```

Comment:
Workaround

Reply:
ah yes I should fix this when we deploy the fix to the indexer but it needs to be tested on the indexer first